src/execution/index/art/art.cpp

```diff
@@ -404,58 +412,68 @@ bool ART::Insert(Node *&node, Key &key, idx_t depth, row_t row_id) {
 	if (!node) {
 		// node is currently empty, create a leaf here with the key
 		node = Leaf::New(key, depth, row_id);
+		this->memory_size += node->MemorySize(*this, false);
 		return true;
 	}
 
 	if (node->type == NodeType::NLeaf) {
-		// Replace leaf with Node4 and store both leaves in it
+		// replace leaf with Node4 and store both leaves in it
 		auto leaf = (Leaf *)node;
-
-		auto &leaf_prefix = leaf->prefix;
 		uint32_t new_prefix_length = 0;
 
-		// Leaf node is already there (its key matches the current key), update row_id vector
+		// FIXME: this code (if and while) can be optimized, less branching, see Construct
+		// leaf node is already there (its key matches the current key), update row_id vector
 		if (new_prefix_length == leaf->prefix.Size() && depth + leaf->prefix.Size() == key.len) {
 			return InsertToLeaf(*leaf, row_id);
 		}
-		while (leaf_prefix[new_prefix_length] == key[depth + new_prefix_length]) {
+		while (leaf->prefix[new_prefix_length] == key[depth + new_prefix_length]) {
 			new_prefix_length++;
-			// Leaf node is already there (its key matches the current key), update row_id vector
+			// leaf node is already there (its key matches the current key), update row_id vector
 			if (new_prefix_length == leaf->prefix.Size() && depth + leaf->prefix.Size() == key.len) {
 				return InsertToLeaf(*leaf, row_id);
 			}
 		}
 
 		Node *new_node = Node4::New();
 		new_node->prefix = Prefix(key, depth, new_prefix_length);
-		auto key_byte = node->prefix.Reduce(new_prefix_length);
-		Node4::InsertChild(new_node, key_byte, node);
+		this->memory_size += new_node->MemorySize(*this, false);
+
+		auto key_byte = node->prefix.Reduce(*this, new_prefix_length);
+		Node4::InsertChild(*this, new_node, key_byte, node);
+
 		Node *leaf_node = Leaf::New(key, depth + new_prefix_length + 1, row_id);
-		Node4::InsertChild(new_node, key[depth + new_prefix_length], leaf_node);
+		Node4::InsertChild(*this, new_node, key[depth + new_prefix_length], leaf_node);
+		this->memory_size += leaf_node->MemorySize(*this, false);
```

Comment:
I'm noticing a pattern of `InsertChild` + `this->memory_size += ...` is there any way we can combine these maybe?